src/schemas/ns10/NicosCacheWriter.h

```diff
@@ -0,0 +1,72 @@
+/** Copyright (C) 2018 European Spallation Source ERIC */
+
+/// \file
+///
+/// \brief Writing module for the NICOS cache values.
+
+#pragma once
+#include "../../FlatbufferReader.h"
+#include "../../HDFWriterModule.h"
+#include "NeXusDataset.h"
+
+#include <iostream>
+
+namespace NicosCacheWriter {
+
+/// h5cpp dataset class that writers strings.
+class StringValue : public NeXusDataset::ExtensibleDataset<std::string> {
+public:
+  StringValue() = default;
+  /// \brief Create the value dataset of NXLog.
+  StringValue(hdf5::node::Group const &Parent, NeXusDataset::Mode CMode,
+              size_t ChunkSize = 1024);
+};
+
+class CacheReader : public FileWriter::FlatbufferReader {
+public:
+  bool verify(FileWriter::FlatbufferMessage const &Message) const override;
+
+  std::string
+  source_name(FileWriter::FlatbufferMessage const &Message) const override;
+
+  uint64_t
+  timestamp(FileWriter::FlatbufferMessage const &Message) const override;
+
+private:
+};
+
+class CacheWriter : public FileWriter::HDFWriterModule {
```

Comment:
Split out this class into another file `Writer.h' so the header files and implementation files match up. E.g.
Reader.h and Reader.cpp
Writer.h and Writer.cpp

Reply:
The reader consists only of a few lines. What about I merge the two?

Reply:
I would prefer them separate like was done for hs00.